ocaml/libs/tracing/tracing.ml

```diff
@@ -87,13 +110,39 @@ module Span = struct
     let context : SpanContext.t = {trace_id; span_id} in
     let begin_time = Unix.gettimeofday () in
     let end_time = None in
-    {context; span_kind; parent; name; begin_time; end_time; tags}
+    let status : Status.t = {status_code= Status.Unset; description= None} in
+    {context; span_kind; status; parent; name; begin_time; end_time; tags}
 
   let finish ?(tags = []) ~span () =
+    let status = span.status in
+    let tags =
+      if status.status_code = Error then
+        (Status.status_to_string status, Printexc.get_backtrace ()) :: tags
+      else
+        tags
+    in
     {span with end_time= Some (Unix.gettimeofday ()); tags= span.tags @ tags}
 
   let set_span_kind span kind = span.span_kind <- kind
 
+  let set_error span exn =
+    let description = Some (Printexc.to_string exn) in
```

Comment:
You need to capture the stacktrace here, before you call anything else (in theory even calling Printexc.to_string might've already destroyed the backtrace).